parcels/particlefile/baseparticlefile.py

```diff
@@ -123,6 +139,41 @@ def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, c
         elif pset_info is None:
             raise IOError("output directory %s already exists. Please remove the directory." % self.tempwritedir)
 
+    def __del__(self):
+        """
+        BaseParticleFile - Destructor
+        """
+        if self.convert_at_end:
+            self.close()
+
+    def __sizeof__(self):
+        """
+        :returns size (in bytes) of the ParticleFile
+        """
+        sz = 0
+        if self.var_names is not None:
+            for vname in self.var_names:
+                sz += sys.getsizeof(getattr(self, vname)) if getattr(self, vname) is not None else 0
+        if self.var_names_once is not None:
+            for vname in self.var_names_once:
+                sz += sys.getsizeof(getattr(self, vname)) if getattr(self, vname) is not None else 0
+        sz += sys.getsizeof(self.outputdt) if self.outputdt is not None else 0
+        sz += sys.getsizeof(self.lasttime_written) if self.lasttime_written is not None else 0
+        sz += sys.getsizeof(self.dataset) if self.dataset is not None else 0
+        sz += sys.getsizeof(self.metadata) if self.metadata is not None else 0
+        sz += sys.getsizeof(self.name) if self.name is not None else 0
+        sz += sys.getsizeof(self.parcels_mesh) if self.parcels_mesh is not None else 0
+        sz += sys.getsizeof(self.time_origin) if self.time_origin is not None else 0
+        sz += sys.getsizeof(self.lonlatdepth_dtype) if self.lonlatdepth_dtype is not None else 0
+        sz += sys.getsizeof(self.var_names) if self.var_names is not None else 0
+        sz += sys.getsizeof(self.file_list) if self.file_list is not None else 0
+        sz += sys.getsizeof(self.var_names_once) if self.var_names_once is not None else 0
+        sz += sys.getsizeof(self.file_list_once) if self.file_list_once is not None else 0
+        sz += sys.getsizeof(self.maxid_written)
+        sz += sys.getsizeof(self.tempwritedir_base) if self.tempwritedir_base is not None else 0
+        sz += sys.getsizeof(self.tempwritedir) if self.tempwritedir is not None else 0
```

Comment:
Would this not be cleaner in a loop?
```python
for var in [self.outputdt, self.lasttime_written]: # and all other variables
    sz += sys.getsizeof(var) if var is not None else 0
```

Reply:
yeah, could be. Did not think of it in my C-code style brain :-)